manifests/010-prometheus-rules.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    name: samples-operator-alerts
+  name: samples-operator-alerts
+  namespace: openshift-cluster-samples-operator
+spec:
+  groups:
+  - name: SamplesOperator
+    rules:
+    - alert: SampleImageStreamImportsFailing
+      # Testing showed that
+      # a) openshift_samples_failed_imagestream_import_total[2h] returned for each imagestream the count at each sample time
+      # b) max_over_time(openshift_samples_failed_imagestream_import_total[2h]) returned a max for each imagestream across all the sample times
+      # c) max(max_over_time(openshift_samples_failed_imagestream_import_total[2h])) returned a single element and max value across all the imagestreams
+      expr: max(max_over_time(openshift_samples_failed_imagestream_import_total[2h])) > 0
```

Comment:
why the max value over 2 hours instead of just the current value?  do we need to fire an alert if 1 hour ago there were some failing imagestreams, but now there are not?

Reply:
my intent was if it persisted over two hours

maybe `max_over_time(sum(openshift_samples_failed_imagestream_import_total[2h]))` ?

I'll experiment some more once I have a cluster up and report, but of course chime in during that interval as your bandwidth permites.

Reply:
i see..yeah i'm not a promql guru so i'm not sure how to get there, but it sounds like maybe you want the min value?  that way if it ever went to zero during those 2 hours, you do *not* throw an alert....only if the min value is 1 (it was failing the entire time) you throw an alert?

`if min_faling_over_2_hour_period > 0 throw alert`

Reply:
Yeah I think replacing `max` with `min` in my existing query does the trick:

`min(min_over_time(openshift_samples_failed_imagestream_import_total[2h]))`

Reply:
check that ... I'm not so sure .... some could be failing and some could succeed ... if so, the min would be 0 for the ones that are succeeding

Reply:
the metric name is `openshift_samples_failed_imagestream_import_total`, is it not the total number of imagestream imports currently failing at a given point in time, across all the imagestreams?

Reply:
Yeah I'm giving up on this one .... we will still mark as degraded if the imports stay failed for two hours, and we will alert on degraded.... essentially the same thing

An admin could then go into prometheus and query on `openshift_samples_failed_imagestream_import_total` to quickly see which imports failed.

Reply:
makes sense to me.

Reply:
yeah perhaps I need to fix the name ... it is really a true/false for whether imports fail on a per imagestream basis.

simply remove the `_total` suffix @bparees ?

Reply:
but I thought it better to convey which imagestreams are failing import

Reply:
I take it that means there is a label on this metric which takes as its value the name of the failing imagestream?

Reply:
i would agree that removing the _total seems appropriate, if so.

Reply:
correct there is a `name` label

will remove _total

Reply:
Per https://prometheus.io/docs/practices/naming/#metric-names and suffix rules I am going to change `_total` to `_info`